pkgs/development/python-modules/setuptools-scm/default.nix

```
@@ -28,6 +28,11 @@ buildPythonPackage rec {
     hash = "sha256-oYOWobwCGcl00adGErEfnc4NW9ix3FXGX2rH/WCejCg=";
   };
 
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail 'tomli<=2.0.2' 'tomli'
+  '';
```

```suggestion
  postPatch = if (pythonOlder "3.11") then ''
    substituteInPlace pyproject.toml \
      --replace-fail 'tomli<=2.0.2' 'tomli'
  '' else null;
```
Last iteration of the fix, I hope :)